### ISSUE TITLE
Audio Desync on Some Streams

### ISSUE THREAD

**Issue by osuegbert1** [TwitchedApp/Twitched#4]
### Description
Some streams have an audio desync of about 2-3 seconds but others work just fine.

### Steps to Reproduce
1. Open app (regular way or with mobile app)
2. Start watching stream

**Expected behavior:** [What was supposed to happen]
Audio and video synced

**Actual behavior:** [What actually happened]
Audio and video not synced

**Roku Model:** [Model]
3600X roku stick

**Roku Firmware Version:** [Firmware version] 
8.0.0 build 4143-24

**Twitched Version:** [Version number]
1.1 build 1745

### Additional Information
This bug did not occur in Twitched version 1.0 so I am assuming it was introduced with version 1.1. I have tried exiting and reloading the stream, restarting twitched, and restarting the roku and the problem still persists for certain streams. I've tried many different streams using different resolutions, framerates, and bitrates and cannot seem to find a pattern.

**Comment by rolandoislas** [comment]
@osuegbert1 is this with chat enabled?

**Comment by osuegbert1** [comment]
I'm not entirely sure what you mean by having chat enabled. The desync happens as soon as I start the stream before i open the chat sidebar if that helps.

**Comment by rolandoislas** [comment]
Possibly caused by #5

**Comment by jbaconbits** [comment]
Yes, this is likely due to lower end hardware that is not spec'd for 1080p60 playback (max should be 720p60 on non 4k models) which causes dropped frames and audio skips.

**Comment by rolandoislas** [comment]
@jbaconbits 25 out of 30 supported Roku models can play at the 1920x1080 resolution. See the [list of current models](https://sdkdocs.roku.com/display/sdkdoc/The+Roku+Channel+Developer+Program#TheRokuChannelDeveloperProgram-CurrentModels) for details.

**Comment by jbaconbits** [comment]
@rolandoislas they’re capable of 1080p @ 30fps, but not at 60fps.

**Comment by rolandoislas** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Quality Autoscaling](https://trello.com/c/fMwNYjzf/27-quality-autoscaling)

**Comment by rolandoislas** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Quality Autoscaling](https://trello.com/c/fMwNYjzf/27-quality-autoscaling)

**Comment by rolandoislas** [comment]
I cannot properly address this without a channel update, so it will be a few weeks before this can be fixed. Would either of you (@osuegbert1, @jbaconbits) be interested in a beta build? If so, please send an email to support@twitched.org and provide an email address attached to a Roku account so it can be added as a test user for Twitched.

**Comment by jbaconbits** [comment]
@rolandoislas Did some testing on my old Roku 3 today and came away confused.

First, the Auto setting on the beta build still seemed to try and pull down a source 1080p60 stream, which caused the dropped frames and audio desync. I tried putting the quality at 720p but still had some issues. I think I ended up needing to restart my Roku for the quality setting to actually stick, but I'm not really sure because a couple times restarting didn't seem to do anything and the channel still tried to pull down the highest quality stream available.

Then I tried the same stream on the current app, and it again pulled down the 1080p60 stream which caused stuttering and audio desync, which I didn't even think was possible on the old app, so I'm not sure if it's because it had something in memory from the beta channel or what.

So...yeah. Not really sure what is going on. Maybe I should delete the current version and power cycle and try again?
